src/components/DatePicker/LocalizedDayPicker.tsx

```diff
@@ -1,22 +1,26 @@
 import clsx from "clsx";
 import React, { forwardRef, useContext, useEffect, useState } from "react";
 import DayPicker, { DayPickerProps } from "react-day-picker";
+import { kebabCase } from "lodash";
 import { Context } from "../Provider";
 import { getLocalizationProps, LocaleCode, LocalizationProps } from "./DatePicker.helpers";
 
 export const LocalizedDayPicker = forwardRef<any, DayPickerProps>(({ className, ...rest }, ref) => {
     const { locale } = useContext(Context);
     const [localizationProps, setLocalizationProps] = useState<Partial<LocalizationProps>>({});
-    console.log("Locale", locale);
 
     useEffect(() => {
         setLocalizationProps({});
+        const localeCode = kebabCase(locale).toLowerCase();
 
         /** We don't want any localization-related props for "English" */
-        if (!locale || locale === "en" || locale === "en_US") return;
+        if (!localeCode || localeCode === "en" || localeCode === "en-us") return;
```

Comment:
we are getting the locale in en-US format, not in en_US format. and english locale object won't have `weekdays` key in the locale obj, so getting the issue `.weekdays` doesn't exist.